applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_3D2N.cpp

```diff
@@ -444,81 +687,91 @@ namespace Kratos
+			typedef array_1d<double, msDimension> arraydim;
+			arraydim global_z = ZeroVector(msDimension);
+			global_z[2] = 1.0;
+
+			arraydim v2 = ZeroVector(msDimension);
+			arraydim v3 = ZeroVector(msDimension);
+
+			double vector_norm;
+			vector_norm = MathUtils<double>::Norm(direction_vector_x);
+			if (vector_norm > numerical_limit) direction_vector_x /= vector_norm;
+
+			if (std::abs(direction_vector_x[2]-1.00) < numerical_limit) {
+				v2[1] = 1.0;
+				v3[0] = -1.0;
+			}
+
+			else if (std::abs(direction_vector_x[2]+1.00) < numerical_limit) {
+				v2[1] = 1.0;
+				v3[0] = 1.0;
+			}
+
+			else {
+				MathUtils<double>::UnitCrossProduct(v2, global_z, direction_vector_x);
+				MathUtils<double>::UnitCrossProduct(v3, direction_vector_x, v2);
+			}
 
-			element_axis.CalculateBasisVectors(direction_vector_x, direction_vector_y,
-				direction_vector_z);
-			element_axis.CalculateRotationMatrix(temp_matrix);
+			//manual rotation around the beam axis
+			if (std::abs(theta_custom) > numerical_limit) {
+				const Vector nz_temp = v3;
+				const Vector ny_temp = v2;
+				const double cos_theta = std::cos(theta_custom);
+				const double sin_theta = std::sin(theta_custom);
+
+				v2 = ny_temp * cos_theta + nz_temp * sin_theta;
+				vector_norm = MathUtils<double>::Norm(v2);
+				if (vector_norm > numerical_limit) v2 /= vector_norm;
+
+				v3 = nz_temp * cos_theta - ny_temp * sin_theta;
+				vector_norm = MathUtils<double>::Norm(v3);
+				if (vector_norm > numerical_limit) v3 /= vector_norm;			
+			}
+
+			for (int i = 0; i < msDimension; ++i) {
+				temp_matrix(i, 0) = direction_vector_x[i];
+				temp_matrix(i, 1) = v2[i];
+				temp_matrix(i, 2) = v3[i];
+			}
 		}
 
-		//save them to update the local axis in every following iter. step
-		this->mNX0 = direction_vector_x;
-		this->mNY0 = direction_vector_y;
-		this->mNZ0 = direction_vector_z;
+		bounded_matrix<double,msElementSize,msElementSize> reference_transformation;
 
 		//Create big rotation Matrix
-		this->AssembleSmallInBigMatrix(temp_matrix, this->mRotationMatrix0);
-
-		//provide Initial Rotation Matrix for strategies that dont call 'CalculateLocalSystem'
-		this->mRotationMatrix = ZeroMatrix(msElementSize);
-		this->mRotationMatrix = this->mRotationMatrix0;
+		this->AssembleSmallInBigMatrix(temp_matrix, reference_transformation);
 
+		return reference_transformation;
 		KRATOS_CATCH("")
 	}
 
 	void CrBeamElement3D2N::CalculateTransformationMatrix(bounded_matrix<double,
-		CrBeamElement3D2N::msElementSize,CrBeamElement3D2N::msElementSize>& rRotationMatrix) {
+		CrBeamElement3D2N::msElementSize,CrBeamElement3D2N::msElementSize>& rRotationMatrix,
+		Vector& Bisectrix, Vector& VectorDifference) {
 
 		KRATOS_TRY
 		//update local CS
 		Matrix aux_rotation_matrix = ZeroMatrix(msDimension);
-		aux_rotation_matrix = this->UpdateRotationMatrixLocal();
+		aux_rotation_matrix = this->UpdateRotationMatrixLocal(Bisectrix,VectorDifference);

```

Comment:
Matrix aux_rotation_matrix =  this->UpdateRotationMatrixLocal(Bisectrix,VectorDifference);